### ISSUE TITLE
how to use model methods in client side?

### ISSUE THREAD

**Issue by bonesoul** [mozilla/nunjucks#974]
So basically I've a mongoose model and it has some helper methods like.

```
CommentSchema.methods = {
  isApproableBy: function (user) {
  ...
  }
};
```

I'm using this comment model browser side via ajax calls;

`var rendered = nunjucks.render('frontend/comments/entry.html', { ajax: true, comment: data }); `

and in my entry.html template, i want to use it;

```
				{% if comment.isApprovableBy(loggedUser) %}
					{% if comment.isRejected %}
						<li><a class="approve">Geri al</a></li>
					{% else %}
						<li style="display: none;"><a class="approve">Geri al</a></li>
					{% endif %}
				{% endif %}
```

So this basically works in server-side but I don't know how to get it working in client side?

**Comment by ArmorDarks** [comment]
It's quite unclear what exactly you're trying to do.

What have you done so far? What exactly doesn't work in client side?

It is very hard to give any specific recommendation, because your example lacks a lot of details and solution to your issue heavily depends on infrastructure.

Few obvious things which might be missing here:

1. In most cases it's good advice to keep most part of logic, especially when it comes to ajax calls, out of templates. It will be much easier to handle them in JavaScript and pass to Nunjucks only result.

   In other words, make any ajax calls and manipulations in JavaScript, then pass results to Nunjucks context and re-render what you need.

2. If you really want to make async ajax calls triggered _from within Nunjucks_ or doing any other async stuff, than you will need to switch Nunjucks to async mode and follow [async templating guidlines](https://mozilla.github.io/nunjucks/api.html#asynchronous-support).

   Otherwise Nunjucks will call render and your template will end up rendered _before_ ajax reply will return.

**Comment by noahlange** [comment]
@bonesoul -

It's unlikely that you'll be able to get this to work out of the box. You'll probably need to recreate your model methods on another class in your front-end code and then "hydrate" it with the data you get via AJAX.

```javascript
```
